### DISCUSSION TITLE
Can Codespaces support other repository like Bitbucket/Gitlab/Azure DevOps rather than Github?

### DISCUSSION THREAD

**Discussion by luxu-ms** [community/community#10479] [Codespaces]
During the conversation with some customers, they are not using Github, but they do use other git repository like Bitbucket/Gitlab/Azure DevOps. Is there any way that the Codespaces can connect to their own git repository rather than Github?

**Comment by asciimike** [comment]
Codespaces are created from a GitHub repo, but once started, you can clone anything you want into them. Here's an example of [cloning an ADO repo into a Codespace](https://github.com/vsls-contrib/ado-in-codespaces), and the process can be repeated for any other platform.
